src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -7643,27 +7643,25 @@ private static void tryFinallyChecks(MethodHandle target, MethodHandle cleanup)
      */
     public static MethodHandle tableSwitch(MethodHandle fallback, MethodHandle... targets) {
         Objects.requireNonNull(fallback);
-        Objects.requireNonNull(targets);
-        targets = targets.clone();
-        MethodType type = tableSwitchChecks(fallback, targets);
-        return MethodHandleImpl.makeTableSwitch(type, fallback, targets);
+        var targetList = List.of(targets); // deep null checks
+        MethodType type = tableSwitchChecks(fallback, targetList);
+        return MethodHandleImpl.makeTableSwitch(type, fallback, targetList);
     }
 
-    private static MethodType tableSwitchChecks(MethodHandle defaultCase, MethodHandle[] caseActions) {
-        if (caseActions.length == 0)
-            throw new IllegalArgumentException("Not enough cases: " + Arrays.toString(caseActions));
+    private static MethodType tableSwitchChecks(MethodHandle defaultCase, List<MethodHandle> caseActions) {
+        if (caseActions.isEmpty())
+            throw new IllegalArgumentException("Not enough cases: " + caseActions);
 
         MethodType expectedType = defaultCase.type();
 
         if (!(expectedType.parameterCount() >= 1) || expectedType.parameterType(0) != int.class)
             throw new IllegalArgumentException(
-                "Case actions must have int as leading parameter: " + Arrays.toString(caseActions));
+                "Case actions must have int as leading parameter: " + caseActions);
```

Comment:
As you're already touching this: `expectedType` is the default case, but it's not printed here. This can lead to very confusing output if all caseActions have a leading int parameter but the defaultCase hasn't.